### DISCUSSION TITLE
Query Path int validation

### DISCUSSION THREAD

**Discussion by bnewman-tech** [fastapi/fastapi#9547] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/{address_id}", response_model=Address)
async def get_address_by_id(
    address_id: int,
    db: AsyncSession = Depends(get_db),
    current_user: PropelAuthUser = Depends(auth.require_user),
):
    """This will return a address based on the ID provided."""
    await get_local_user_and_check_role(
        propel_user=current_user,
        db=db,
        required_permissions=[AuthRoles.can_view_addresses],
    )
    results = await crud_address.get_by_id_joined(db=db, id=address_id)
    if results is None:
        raise HTTPException(
            status_code=404,
            detail=f"Address with ID of {address_id} was not found",
        )
    return results
```

### Description

I have this endpoint definition. When I call the database I am getting an error that int in out of int32 range. 
Is there a way to check for that error on the path field (address_id: int) or the best way to check that number to be less than or equal to 2,147,483,647? 

Thank you for your help!

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.11

**Comment by nymous** [comment, marked as answer]
Hello!
Your problem seems to be solved exactly by this docs page: https://fastapi.tiangolo.com/tutorial/path-params-numeric-validations/#number-validations-greater-than-and-less-than-or-equal.
For example,
```py
from typing import Annotated

from fastapi import Path

@router.get("/{address_id}", response_model=Address)
async def get_address_by_id(
    address_id: Annotated[int, Path(le=2_147_483_647)],  # or Annotated[int, Path(le=(2**31) - 1)]
    db: AsyncSession = Depends(get_db),
    current_user: PropelAuthUser = Depends(auth.require_user),
):
```

**Comment by bnewman-tech** [comment]
Thank you so much for taking the time to assist me! That is perfect! 
Sorry I missed that! I have not used Annotated yet but like the concept.
